app/controllers/pages/questions_controller.rb

```diff
@@ -0,0 +1,83 @@
+class Pages::QuestionsController < PagesController
+  def new
+    answer_type = session.dig(:page, :answer_type)
+    question_text = session.dig(:page, :question_text)
+    answer_settings = session.dig(:page, :answer_settings)
+    is_optional = session.dig(:page, :is_optional) == "true"
+    page_heading = session.dig(:page, :page_heading)
+    guidance_markdown = session.dig(:page, :guidance_markdown)
+    @question_form = Pages::QuestionForm.new(form_id: @form.id, answer_type:, question_text:, answer_settings:, is_optional:)
+    @page = Page.new(form_id: @form.id, question_text:, answer_type:, answer_settings:, is_optional:, page_heading:, guidance_markdown:)
+
+    render "pages/new"
+  end
+
+  def create
+    answer_settings = session.dig(:page, :answer_settings)
+    page_heading = session.dig(:page, :page_heading)
+    guidance_markdown = session.dig(:page, :guidance_markdown)
+
+    @page = Page.new(page_params.merge(answer_settings:, page_heading:, guidance_markdown:))
+
+    @question_form = Pages::QuestionForm.new(page_params.merge(answer_settings:, page_heading:, guidance_markdown:))
```

Comment:
Teensy tiny formatting suggestion, right now with the line breaks where they are I find it easy to miss the if statement, I think this would read better:

```suggestion
    @page = Page.new(page_params.merge(answer_settings:, page_heading:, guidance_markdown:))
    @question_form = Pages::QuestionForm.new(page_params.merge(answer_settings:, page_heading:, guidance_markdown:))

```